sui/typescript/backend_examples/src/gas_station.ts

```diff
@@ -93,6 +94,46 @@ async function clockMoveCallGaslessTransaction(): Promise<GaslessTransaction> {
   return gaslessTx
 }
 
+//
+// Builds a Move call transaction for sponsorship in multiple steps with the following behaviors:
+//  1. Manually budgets the sponsorship (no Shinami auto-budgeting, with its sui_dryRunTransactionBlock
+//   for gas budget estimation and its ability to catch some transaction errors pre-sponsorship.)
+//   We recommend most integrations use auto-budgeting to be safe, but if you need the lowest latency 
+//    possible, especially if you already make a sui_dryRunTransactionBlock request, manual budgeting 
+//    makes sense.
+//  2. Building the transaction offline: https://docs.sui.io/guides/developer/sui-101/building-ptb#building-offline
+//      This can save a few requests that would otherwise get made: https://docs.shinami.com/docs/transaction-block-build-requests
+//      Note through that those requests are extremely fast for us to fulfill once they hit our servers, so this is only a latency 
+//      concern if your servers are geographically far from one of our Fullnodes. 
```

Comment:
Both of these behaviors should be supported by the `buildGaslessTransaction` helper. I wonder why you chose to do it in this example.

Reply:
When I was initially writing it out I wanted to make the steps very explicit. I've simplified it to use `buildGaslessTransaction` and added a comment to call out the hidden call to Transaction.build()